orttraining/orttraining/python/training/torchdynamo/ort_backend.py

```diff
@@ -0,0 +1,474 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from typing import Any, Callable, Dict, List, Mapping, Set, Tuple, Type
+
+import numpy as np
+import onnx
+import torch
+import torch._prims.executor
+import torch.onnx._onnx_supported_ops
+from torch._decomp import decomposition_table
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.fx.passes.fake_tensor_prop import FakeTensorProp
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
+from torch.fx.passes.operator_support import OperatorSupport
+from torch.fx.passes.tools_common import CALLABLE_NODE_OPS
+from torch.onnx._globals import GLOBALS as ONNX_GLOBALS
+
+import onnxruntime  # type: ignore
+from onnxruntime.capi import _pybind_state as ORTC
+
+_NP_DTYPE = {
+    torch.float16: np.float16,
+    torch.float32: np.float32,
+    torch.float64: np.float64,
+    torch.uint8: np.uint8,
+    torch.int8: np.int8,
+    torch.int16: np.int16,
+    torch.int32: np.int32,
+    torch.int64: np.longlong,
+    torch.bool: np.bool_,
+}
```

Comment:
Probably not because of

1. it will introduce `libtorch` dependency into ORT's pybind code.
2. wrapping light-weight Python as C++ code usually reduces performance.